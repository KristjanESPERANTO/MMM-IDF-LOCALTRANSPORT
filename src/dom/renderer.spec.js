/* @flow */

import {
  renderWrapper,
  renderHeader,
  renderTraffic,
  renderNoInfoVelib,
  renderVelib,
} from './renderer.js';

describe('renderWrapper function', () => {
  it('should return correct HTML when not loaded', () => {
    // given-when
    const actual = renderWrapper(false);
    // then
    expect(actual.outerHTML).toMatchSnapshot();
  });

  it('should return correct HTML when loaded', () => {
    // given-when
    const actual = renderWrapper(true);
    // then
    expect(actual.outerHTML).toMatchSnapshot();
  });
});

describe('renderHeader function', () => {
  const baseConfig = {
    updateInterval: 60000,
    lastUpdate: new Date(),
    showLastUpdateTime: false,
    showSecondsToNextUpdate: false,
  };

  it('should return empty string when empty configuration', () => {
    // given
    const config = {};
    // when
    const actual = renderHeader(config);
    // then
    expect(actual).toEqual('');
  });

  it('should return correct header when complete configuration', () => {
    // given
    const config = Object.assign({}, baseConfig, { showLastUpdateTime: true, showSecondsToNextUpdate: true });
    // when
    const actual = renderHeader(config);
    // then
    expect(actual).toContain(', next update in ');
    expect(actual).toContain('@');
  });

  it('should return correct header when incomplete configuration 1', () => {
    // given
    const config = Object.assign({}, baseConfig, { showSecondsToNextUpdate: true });
    // when
    const actual = renderHeader(config);
    // then
    expect(actual).toContain(', next update in ');
    expect(actual).not.toContain('@');
  });

  it('should return correct header when incomplete configuration 2', () => {
    // given
    const config = Object.assign({}, baseConfig, { showLastUpdateTime: true });
    // when
    const actual = renderHeader(config);
    // then
    expect(actual).not.toContain(', next update in ');
    expect(actual).toContain('@');
  });

  it('should return empty string when silent configuration', () => {
    // given-when
    const actual = renderHeader(baseConfig);
    // then
    expect(actual).toEqual('');
  });
});

describe('renderTraffic function', () => {
  it('should return correct HTML for table row', () => {
    // given
    const stop = {
      line: ['BUS', 275],
      label: 'Ulbach',
      stations: '',
    };
    const ratpTraffic = {
      'traffic/bus/275': {
        message: 'fluid',
      },
    };
    const config = {
      conversion: {
        fluid: 'fluide',
      },
    };
    // when
    const actual = renderTraffic(stop, ratpTraffic, config);
    // then
    expect(actual.outerHTML).toMatchSnapshot();
  });
});

describe('renderNoInfoVelib function', () => {
  it('should return correct HTML for table row when label', () => {
    // given
    const stop = {
      line: ['VELIB', 68],
      label: 'Ulbach',
      stations: '',
    };
    // when
    const actual = renderNoInfoVelib(stop);
    // then
    expect(actual.outerHTML).toMatchSnapshot();
  });

  it('should return correct HTML for table cell when no label', () => {
    // given
    const stop = {
      line: ['VELIB', 68],
      stations: 'Stations',
    };
    // when
    const actual = renderNoInfoVelib(stop);
    // then
    expect(actual.outerHTML).toMatchSnapshot();
  });
});

describe('renderVelib function', () => {

  it('should return correct HTML when no history', () => {
    // given
    const stop = {
      line: ['VELIB', 68],
      label: 'Ulbach',
      stations: '',
    };
    const velibHistory = {};
    const config = {};
    const now = new Date();
    // when
    const actual = renderVelib(stop, velibHistory, config, now);
    // then
    expect(actual.outerHTML).toMatchSnapshot();
  });

  it('should return correct HTML when history without trend', () => {
    // given
    const stop = {
      line: ['VELIB', 68],
      stations: 'Stations',
    };
    const velibHistory = {
      Stations: [{
        total: 10,
        bike: 2,
        empty: 8,
        name: 'Opera',
      }],
    };
    const config = {
      trendGraphOff: true,
    };
    const now = new Date(2017, 5, 29, 8, 34, 28);
    // when
    const actual = renderVelib(stop, velibHistory, config, now);
    // then
    expect(actual.outerHTML).toMatchSnapshot();
  });

  it('should return correct HTML when history with trend', () => {
    // given
    const stop = {
      line: ['VELIB', 68],
      stations: 'Stations',
    };
    const velibHistory = {
      Stations: [{
        total: 10,
        bike: 2,
        empty: 8,
        name: 'Opera',
      }],
    };
    const config = {};
    const now = new Date(2017, 5, 29, 8, 34, 28);
    // when
    const actual = renderVelib(stop, velibHistory, config, now);
    // then
    expect(actual.outerHTML).toMatchSnapshot();
  });
});
