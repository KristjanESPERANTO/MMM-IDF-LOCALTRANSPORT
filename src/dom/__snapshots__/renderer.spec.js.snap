// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderNoInfoVelib function should return correct HTML for table cell when no label 1`] = `"<tr><td class=\\"bright\\">Stations no info yet</td></tr>"`;

exports[`renderNoInfoVelib function should return correct HTML for table row when label 1`] = `"<tr><td class=\\"bright\\">Ulbach no info yet</td></tr>"`;

exports[`renderTraffic function should return correct HTML for table row 1`] = `"<tr><td class=\\"align-right bright\\">Ulbach</td><td class=\\"align-left\\" colspan=\\"2\\">fluide</td></tr>"`;

exports[`renderVelib function should return correct HTML when history with trend 1`] = `"<tr></tr>"`;

exports[`renderVelib function should return correct HTML when history without trend 1`] = `"<tr><td class=\\"align-left\\">10</td><td class=\\"bright\\">2 velibs 8 spaces</td><td class=\\"align-right\\">Opera</td></tr>"`;

exports[`renderVelib function should return correct HTML when no history 1`] = `"<tr><td class=\\"bright\\">Ulbach no info yet</td></tr>"`;

exports[`renderWrapper function should return correct HTML when loaded 1`] = `"<div class=\\"paristransport\\"></div>"`;

exports[`renderWrapper function should return correct HTML when not loaded 1`] = `"<div class=\\"dimmed light small\\">Loading connections ...</div>"`;
